src/sentry/workflow_engine/types.py

```diff
@@ -46,7 +50,7 @@ class DetectorEvaluationResult:
     is_triggered: bool
     priority: DetectorPriorityLevel
     # TODO: This is only temporarily optional. We should always have a value here if returning a result
-    result: IssueOccurrence | StatusChangeMessage | None = None
+    result: IssueOccurrence | StatusChangeMessage | DetectorOccurrence | None = None
```

Comment:
This is having an effect on some downstream types - wanna make sure this change is ok before I change those too. We emit a `DetectorOccurrence` that later gets changed to an `IssueOccurrence` via `to_issue_occurrence` because we generate the fingerprint, detection time, etc outside of creating the detector occurrence. I'm not actually sure if we have handlers emitting `IssueOccurrence` types directly, do we?

Reply:
if possible, i'd prefer to not have the `detector occurrence` here, that way we can ensure the result is always something that can be sent to the issue platform. the hope is that this is just the final thing that we produce in `process_detectors`, and then it can be sent to the issue platform.

are we asking for a `DetectorEvaluationResult` in one of the handler methods? (for some reason i thought that just wanted a detector occurrence)

Reply:
Ugh this was actually a desperate attempt to make typing pass in my tests